modules/library/xml/src/main/java/org/geotools/gml/producer/CoordinateFormatter.java

```diff
@@ -99,7 +99,7 @@ private String repeatZeros(int num) {
     /** Formats a number with the configured number of decimals */
     public StringBuffer format(double x, StringBuffer sb) {
         String formatted;
-        if ((Math.abs(x) >= DECIMAL_MIN && x < DECIMAL_MAX) || x == 0) {
+        if ((Math.abs(x) >= DECIMAL_MIN && Math.abs(x) < DECIMAL_MAX) || x == 0) {
```

Comment:
While this logic is correct, it does show an issue with truncate(x) below. 

If I understand correctly the following is incorrect:
```
scaled += 0.5;
```

And should be:

```
scaled += Math.signnum(x) * 0.5;
```

Still the above feedback goes a bit beyond what your PR offered to fix.

Reply:
I didn't delve into truncate. How did you notice the issue?

Reply:
I looked at testFormatScientificDECIMAL_MIN and noticed you did not test negative numbers, so I wondered if it would work and reviewed truncate.

My understanding may be wrong fo course.

Reply:
I would / will make that change, but I'm not at my dev machine (for about a week).